test/Common/HttpApiHandlerTests.cs

```diff
@@ -836,6 +836,140 @@ public async Task StartNewInstance_Returns_HTTP_400_On_Missing_Function()
             Assert.Equal(exceptionMessage, error["ExceptionMessage"].ToString());
         }
 
+        [Theory]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task GetEntity_Returns_State_Or_HTTP_404(bool hasKey, bool exists)
+        {
+            string entity = "SomeEntity";
+            string key = hasKey ? Guid.NewGuid().ToString("N") : "";
+            var uriBuilder = new UriBuilder(TestConstants.NotificationUrl);
+
+            uriBuilder.Path += $"/entities/{entity}";
+
+            if (!string.IsNullOrEmpty(key))
+            {
+                uriBuilder.Path += $"/{key}";
+            }
+
+            var testRequest = new HttpRequestMessage
+            {
+                Method = HttpMethod.Get,
+                RequestUri = uriBuilder.Uri,
+            };
+
+            var entityId = new EntityId(entity, key);
+            var result = new EntityStateResponse<JToken>() { EntityExists = exists, EntityState = exists ? new JObject() : null };
+            var clientMock = new Mock<IDurableOrchestrationClient>(MockBehavior.Strict);
+
+            clientMock
+                    .Setup(x => x.ReadEntityStateAsync<JToken>(entityId, null, null))
+                    .Returns(Task.FromResult(result));
+
+            var httpApiHandler = new ExtendedHttpApiHandler(clientMock.Object);
+            var actualResponse = await httpApiHandler.HandleRequestAsync(testRequest);
+
+            if (exists)
+            {
+                Assert.Equal(HttpStatusCode.OK, actualResponse.StatusCode);
+
+                var content = await actualResponse.Content.ReadAsStringAsync();
+                Assert.Equal("{}", content);
+            }
+            else
+            {
+                Assert.Equal(HttpStatusCode.NotFound, actualResponse.StatusCode);
+            }
+        }
+
+        [Theory]
+        [InlineData(false, false, false)]
+        [InlineData(false, false, true, true)]
+        [InlineData(false, false, true, false)]

```

Comment:
Why do some tests rely on the default value of `hasJsonContent` and others explicitly set it to `false`?

Reply:
It is meant to convey that the the fourth argument (`hasJsonContent`) is irrelevant unless the third argument (`hasContent`) is true.